File: src/arrays/min_rotated_array.py
from typing import List


class Solution:
    def findMin(self, nums: List[int]) -> int:
        leftPointer = 0
        rightPointer = len(nums) - 1
        while leftPointer <= rightPointer:
            if nums[leftPointer] <= nums[rightPointer]:
                break
            middlePointer = int((leftPointer + rightPointer) / 2)
            if nums[middlePointer] >= nums[leftPointer]:
                leftPointer = middlePointer + 1
            else:
                rightPointer = middlePointer
        return nums[leftPointer]

File: java/src/main/java/com/blind75/arrays/MinRotatedArray.java
package com.blind75.arrays;

public class MinRotatedArray {

  public int findMin(int[] nums) {
    var leftPointer = 0;
    var rightPointer = nums.length - 1;
    while (leftPointer <= rightPointer) {
      var middle = (rightPointer + leftPointer) / 2;
      if (nums[leftPointer] <= nums[rightPointer]) {
        break;
      }
      if (nums[middle] >= nums[leftPointer]) {
        leftPointer = middle + 1;
      } else {
        rightPointer = middle;
      }
    }
    return nums[leftPointer];
  }

}
